### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by kberzinch**
Key features:
- Route 53 health checks used to hit `/ping` endpoint every 30 seconds from each of 3 regions
- If more than 5 failures occur then DNS will fail over to CloudFront (and fail back when 5 successful responses are received)
- CloudFront distribution returns 503 and an HTML error page for any URL except `/maintenance-page.html`

### CODE REVIEW

**Review by JoshOldenburg — APPROVED** [review]

File: cloudfront.tf [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  default_root_object = "index.html" # this is deliberately a nonexistent object so cloudfront returns a 503
+  enabled             = true
+  is_ipv6_enabled     = false
+  price_class         = "PriceClass_100"
+
+  restrictions {
+    geo_restriction {
+      locations        = ["US"]
```
Comment (evan10s) [inline cloudfront.tf]: Is there a reason for this restriction?  There are times (study abroad, international student at home, etc.) where RJ members might try to access MyRoboJackets from outside the US

Comment (kberzinch) [reply cloudfront.tf]: It's mostly intended as a cost-saving measure, as _most_ clients are in the US. I can take it out.

Do you also want to change the price class? [[docs](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/PriceClass.html)]

Comment (evan10s) [reply cloudfront.tf]: It sounds like PriceClass_100 without the geographic restriction (if I'm understanding correctly) would just cause higher latency outside the US and Europe and that seems fine to me

Comment (kberzinch) [reply cloudfront.tf]: That's correct.

**Review by evan10s — APPROVED** [review]

**Pull request merged**